### PR TITLE
Create email confirmation logic

### PR DESCRIPTION
**Pull request by Eugene-pas**

### CODE REVIEW

File: src/components/emailConfirmation/index.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            <div className="authBody">
+            <div className="center">
+
+                <p className="title">Logistman</p>
+
+                <h2 className="h2ConfirmEmail" >Your email has been </h2>
+
+                <h1 className="confirm" >confirmed</h1>
+
+
```
Comment (Grasmus) [inline src/components/emailConfirmation/index.js]: Remove empty line

File: src/components/emailConfirmation/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import React, {useEffect, useState} from 'react';
+import {confirmEmail, confirmEmailAsync} from "../../services/authentication";
+
+function ConfirmEmail(){
+    let isConfirm = true;
```
Comment (Grasmus) [inline src/components/emailConfirmation/index.js]: Why is it true by default?

File: src/components/emailConfirmation/index.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        let email = data[data.length - 1];
+
+        isConfirm = await confirmEmailAsync(email, token).then((result) =>{
+            return result;
+        })
+
+        setIsConfirmData(isConfirm)
+    });
+
+    if(isConfirmData){
```
Comment (Grasmus) [inline src/components/emailConfirmation/index.js]: You can write if statement only in the place, where you have changes

File: src/assets/scss/assets/pages/_emailConfirmation.scss [hunk 4]
```diff
@@ -0,0 +1,14 @@
+@import '../variables/colors';
+@import '../variables/others';
+
+.h2ConfirmEmail {
```
Comment (marynakernychna) [inline src/assets/scss/assets/pages/_emailConfirmation.scss]: Wrap styles in a separate class for this component. Otherwise, they will be global

Comment (marynakernychna) [inline src/assets/scss/assets/pages/_emailConfirmation.scss]: Rename this class name

File: src/components/emailConfirmation/index.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import React, {useEffect} from 'react';
+import {confirmEmailAsync} from "../../services/authentication";
+
+function ConfirmEmail() {
+
```
Comment (marynakernychna) [inline src/components/emailConfirmation/index.js]: Remove the extra empty line

File: src/components/emailConfirmation/index.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import React, {useEffect} from 'react';
+import {confirmEmailAsync} from "../../services/authentication";
+
+function ConfirmEmail() {
```
Comment (marynakernychna) [inline src/components/emailConfirmation/index.js]: Rename to ConfirmEmailPage

File: src/components/emailConfirmation/index.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+    useEffect(async () => {
+        let data = window.location.pathname.split('/');
+        let tokenParts = data.slice(2, -1);
+        let token = tokenParts.join('/');
+
+        await confirmEmailAsync(token);
+    });
```
Comment (marynakernychna) [inline src/components/emailConfirmation/index.js]: Add a square brackets

File: src/components/emailConfirmation/index.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+    return (
+        <div className="authBody">
```
Comment (marynakernychna) [inline src/components/emailConfirmation/index.js]: This is not the body of the authentication page

File: src/components/emailConfirmation/index.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+            <div className="center">
+
+                <p className="title">Logistman</p>
```
Comment (marynakernychna) [inline src/components/emailConfirmation/index.js]: Why not the h1 tag or something like that

File: src/components/emailConfirmation/index.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+                <h2 className="h2ConfirmEmail">Your email has been </h2>
```
Comment (marynakernychna) [inline src/components/emailConfirmation/index.js]: Rename the class name

Comment (marynakernychna) [inline src/components/emailConfirmation/index.js]: Remove the extra space

File: src/components/emailConfirmation/index.js [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+                <h1 className="confirm">confirmed</h1>
```
Comment (marynakernychna) [inline src/components/emailConfirmation/index.js]: Very strange breakdown of the text

File: src/components/emailConfirmation/index.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+
```
Comment (marynakernychna) [inline src/components/emailConfirmation/index.js]: Remove the extra empty line

File: src/components/emailConfirmation/index.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+            </div>
+        </div>);
```
Comment (marynakernychna) [inline src/components/emailConfirmation/index.js]: Move the bracket to a new line

File: src/constants/messages/authentication.js [hunk 14]
```diff
@@ -9,5 +9,8 @@ export const authenticationErrorMessages = Object.freeze({
     REGISTRATION_FAILED_USER_ALREADY_EXIST:
         "There is already a user with this e-mail!",
 
+    SEND_EMAIL_CONFIRMATION_FAILED: "Send emailConfirmation confirmation failed!",
```
Comment (marynakernychna) [inline src/constants/messages/authentication.js]: Rewrite the message

File: src/constants/messages/general.js [hunk 15]
```diff
@@ -1,5 +1,6 @@
 export const generalErrorMessages = Object.freeze({
     CORRECT_ALL_COMMENTS: "Please correct all comments first!",
     SOMETHING_WENT_WRONG: "Something went wrong, try again!",
-    CHANGE_SOMETHING_TO_SAVE: "Change something to save!"
+    CHANGE_SOMETHING_TO_SAVE: "Change something to save!",
+    EMAIL_CONFIRM: "Your mail has already been confirmed or the confirmation time has expired."
```
Comment (marynakernychna) [inline src/constants/messages/general.js]: What does this do in this file? It must be in another

File: src/services/authentication.js [hunk 16]
```diff
@@ -119,3 +119,28 @@ export function checkIsUserRoleValid() {
...
     }
 }
+
+export async function confirmEmailAsync(token){
+    let model = {
+        token: token
+    }
+
+    return authenticationService
+        .confirmEmail(model)
```
Comment (marynakernychna) [inline src/services/authentication.js]: You can simply send a token without intermediate model creation

File: src/constants/messages/general.js (hunk 15 above)
Comment (marynakernychna) [inline src/constants/messages/general.js]: Rename the constant

**Review by marynakernychna — CHANGES_REQUESTED** [review]

**Review by marynakernychna — COMMENTED** [review]
Resolve merge conflicts

File: src/components/emailConfirmation/index.js [hunk 17]
```diff
@@ -0,0 +1,28 @@
+import React, {useEffect} from 'react';
```
Comment (marynakernychna) [inline src/components/emailConfirmation/index.js]: index.jsx

File: src/App.js [hunk 18]
```diff
@@ -16,8 +16,10 @@ import MyOffersPage from './components/myOffers/index';
 import RegistrationPage from './components/authentication/registration/index';
 import LoginPage from './components/authentication/login/index';
 import HomePage from './components/home/index';
+import ConfirmEmailPage from "./components/emailConfirmation";
 import ManageUsersPage from './components/users/index';
 
+
```
Comment (Grasmus) [inline src/App.js]: Remove empty line

File: src/assets/scss/main.scss [hunk 19]
```diff
@@ -3,8 +3,10 @@
 @import "./assets/pages/profile";
 @import "./assets/pages/userCars";
 @import "./assets/pages/userOffers";
+@import "./assets/pages/emailConfirmation";
 @import "./assets/pages/users";
 
+
```
Comment (Grasmus) [inline src/assets/scss/main.scss]: Remove empty line

File: src/components/emailConfirmation/index.jsx [hunk 20]
```diff
@@ -0,0 +1,26 @@
...
+
+        await confirmEmailAsync(token);
+    }, []);
+
+    return (
+        <div className="authEmail">
+            <div className="center">
+                <h1 className="title">Logist</h1>
+
+                <h2>Your email has been</h2>
```
Comment (Grasmus) [inline src/components/emailConfirmation/index.jsx]: Why is it separated from other containers?

**Review by marynakernychna — APPROVED** [review]

**Review by Grasmus — APPROVED** [review]

**Pull request merged**